<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1002</width>
    <height>702</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="ShapeRenderer" name="shapeRenderer" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>581</width>
      <height>701</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>0</y>
      <width>411</width>
      <height>701</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,0,0,0,0,0,0,0,0,0,0,0">
     <property name="leftMargin">
      <number>10</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>10</number>
     </property>
     <property name="bottomMargin">
      <number>3</number>
     </property>
     <item>
      <widget class="Line" name="line_6">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_18">
       <property name="text">
        <string>Shape</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox">
       <item>
        <property name="text">
         <string>Square</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Rectangle</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Rhombus</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Hexagon</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>N-Star</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Random Polygon</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Circle</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Triangle</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Drawing</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <widget class="QStackedWidget" name="stackedWidget">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="squarePagre">
        <widget class="QWidget" name="gridLayoutWidget_2">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>10</y>
           <width>371</width>
           <height>71</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,0,0" columnminimumwidth="0,0,0">
          <item row="0" column="2">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>px</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="sideValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="side">
            <property name="text">
             <string>Side:</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="rectanglePage">
        <widget class="QWidget" name="gridLayoutWidget">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>0</y>
           <width>331</width>
           <height>111</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="sideA">
            <property name="text">
             <string>Side A:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="sideAValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="pxA">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="sideB">
            <property name="text">
             <string>Side B:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="sideBValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="pxB">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="rhombusPage">
        <widget class="QWidget" name="gridLayoutWidget_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>0</y>
           <width>371</width>
           <height>91</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_7">
          <item row="0" column="1">
           <widget class="QSpinBox" name="rhombusSideValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>Side:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>Acute angle:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="rhombusAngleValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="hexagonPage">
        <widget class="QWidget" name="gridLayoutWidget_3">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>381</width>
           <height>91</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="1">
           <widget class="QSpinBox" name="hexagonRadiusValue">
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>Radius:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="starPage">
        <widget class="QWidget" name="gridLayoutWidget_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>0</y>
           <width>371</width>
           <height>91</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="text">
             <string>Number:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="NValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>64</number>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Ratio:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="ratioValue">
            <property name="minimum">
             <double>1.300000000000000</double>
            </property>
            <property name="maximum">
             <double>4.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.200000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="weirdStarPage">
        <widget class="QWidget" name="gridLayoutWidget_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>0</y>
           <width>371</width>
           <height>91</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_8">
          <item row="0" column="0">
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>Number:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="weirdNValue">
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>64</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="circlePage">
        <widget class="QWidget" name="gridLayoutWidget_7">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>391</width>
           <height>91</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="0" column="1">
           <widget class="QSpinBox" name="circleRadiusValue">
            <property name="singleStep">
             <number>10</number>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Radius:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="trianglePage">
        <widget class="QWidget" name="gridLayoutWidget_8">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>391</width>
           <height>141</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="1">
           <widget class="QSpinBox" name="triangleAValue">
            <property name="maximum">
             <number>500</number>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_27">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>Side B:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>Side A:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_29">
            <property name="text">
             <string>Side C:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSpinBox" name="triangleBValue">
            <property name="maximum">
             <number>500</number>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="triangleCValue">
            <property name="maximum">
             <number>500</number>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="label_30">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <widget class="QLabel" name="label_31">
            <property name="text">
             <string>px</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QWidget" name="page"/>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>Info</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_5" rowstretch="0,0,0" rowminimumheight="0,0,0">
       <property name="sizeConstraint">
        <enum>QLayout::SetDefaultConstraint</enum>
       </property>
       <property name="topMargin">
        <number>17</number>
       </property>
       <property name="bottomMargin">
        <number>25</number>
       </property>
       <property name="verticalSpacing">
        <number>50</number>
       </property>
       <item row="0" column="4">
        <widget class="QLabel" name="angleText">
         <property name="font">
          <font>
           <family>Noto Traditional Nushu</family>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="squareValue">
         <property name="font">
          <font>
           <family>Noto Traditional Nushu</family>
           <bold>true</bold>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QLabel" name="coordY">
         <property name="font">
          <font>
           <family>Noto Traditional Nushu</family>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="perimeterValue">
         <property name="font">
          <font>
           <family>Noto Traditional Nushu</family>
           <bold>true</bold>
          </font>
         </property>
         <property name="frameShape">
          <enum>QFrame::NoFrame</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Plain</enum>
         </property>
         <property name="lineWidth">
          <number>3</number>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="squareText">
         <property name="text">
          <string>Square:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QLabel" name="scaleText">
         <property name="font">
          <font>
           <family>Noto Traditional Nushu</family>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Angle:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QLabel" name="coordX">
         <property name="font">
          <font>
           <family>Noto Traditional Nushu</family>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="perimeterText">
         <property name="text">
          <string>Perimeter:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Scale:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>X:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="coordinatesText">
         <property name="text">
          <string>Center:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="3">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Y:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="line_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_16">
       <property name="text">
        <string>Transfomration</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="0" column="2">
        <widget class="QSpinBox" name="moveToXTransform">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Anchor point:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QSpinBox" name="anchorYTransform">
         <property name="singleStep">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QDoubleSpinBox" name="scaleTransform">
         <property name="singleStep">
          <double>0.200000000000000</double>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QSpinBox" name="moveToYTransform">
         <property name="minimum">
          <number>-500</number>
         </property>
         <property name="maximum">
          <number>500</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>X:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSpinBox" name="angleTransform">
         <property name="singleStep">
          <number>12</number>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSpinBox" name="anchorXTransform">
         <property name="singleStep">
          <number>10</number>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Y</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>Angle:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>X:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>Move to:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Scale:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="4">
        <widget class="QPushButton" name="resetButton">
         <property name="text">
          <string>Reset Transform</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ShapeRenderer</class>
   <extends>QWidget</extends>
   <header>shaperenderer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>comboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>793</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>1041</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
